src/filter.jl

```diff
@@ -150,79 +140,58 @@ function signatures_from_hessian(hes::AbstractArray{T,4}) where T
-                cats[i,j,k, 2] = (abs(eigs[2] / eigs[1]) *
-                    xθ(1-abs(eigs[3] / eigs[1]))) *
-                    (abs(eigs[2]) * θ(-eigs[1]) * θ(-eigs[2]) )
-                cats[i,j,k, 3] = (xθ(1-abs(eigs[2] / eigs[1])) *
-                    xθ(1-abs(eigs[3] / eigs[1]))) *
-                    (abs(eigs[1]) * θ(-eigs[1]))
-                
-                sigs[i,j,k,1] = cats[i,j,k, 1] * abs(eigs[3])*θ(-eigs[1])*θ(-eigs[2])*θ(-eigs[3])
-                sigs[i,j,k,2] = cats[i,j,k, 2] * abs(eigs[2])*θ(-eigs[1])*θ(-eigs[2])
-                sigs[i,j,k,3] = cats[i,j,k, 3] * abs(eigs[1])*θ(-eigs[1])
+                e1, e2, e3 = sort(real(eigvals(hes_slice)), rev=false)
+
+                sigs[i,j,k,1] = (abs(e3 / e1) *
+                    abs(e3)) * (θ(-e1) * θ(-e2) * θ(-e3))
+                sigs[i,j,k,2] = (abs(e2 / e1) *
+                    xθ(1-abs(e3 / e1))) * (abs(e2) * θ(-e1) * θ(-e2) )
+                sigs[i,j,k,3] = (xθ(1-abs(e2 / e1)) *
+                    xθ(1-abs(e3 / e1))) * (abs(e1) * θ(-e1))
             end
         end
-        # GC.gc()
     end
-    #return cats  # category shape strength ℐ (arxiv:1209.2043 eq 6)
+
+    # category shape strength 𝒮 (arxiv:1209.2043 eq 7)
     return sigs
 end
 
 
-function maximum_signature(R_0::Float64, field::AbstractArray{T,3}) where T
-    """
-    In this function we start at minimum length scale to calculate signatures
-    then iterate to the maximum scale of the problem, calculating the signatures
-    for each smoothing scale. We then chose the maximum signature over all scales
-    at each point to categorize whether that point is a wall, filament, or cluster.
+"""
+In this function we start at minimum length scale to calculate signatures
+then iterate to the maximum scale of the problem, calculating the signatures
+for each smoothing scale. We then chose the maximum signature over all scales
+at each point to categorize whether that point is a wall, filament, or cluster.
+"""
+function maximum_signature(Rs, field::AbstractArray{T,3}; alg=:NEXUSPLUS) where T
 
-    R_0 is typically the grid spacing of the field
-    """
+    if alg ∉ (:NEXUS, :NEXUSPLUS)
+        throw(ArgumentError("alg must be either :NEXUS or :NEXUSPLUS"))
+    end
 
     nx,ny,nz = size(field)
 
-    #create an array of scales
-    Rs = []
-    n::Int64 = 5 #this will need to be changed
-    for i in 1:n
-        Ri = R_0 * sqrt(2)^i
-        Rs = append!(Rs,Ri)
-    end
-
     #calculate wave vectors for our field
-    wave_vecs = wavevectors3D((nx,ny,nz))
+    wave_vecs = wavevectors3D(T, (nx,ny,nz))
 
-    #calculate signatures at each scale Rn then determine the max signature
-    sigs_ = zeros(T,nx,ny,nz,3,length(Rs))
-    sigs_final = zeros(T,nx,ny,nz,3)
+    #calculate signatures at each scale Rn, determine max
+    sigmax = zeros(T,nx,ny,nz,3)
 
-    for a in 1:length(Rs)
-        log_smooth_field = smooth_loggauss(field, Rs[a], wave_vecs)
```

Comment:
This was previously smoothing twice. Once in line 205, and once in the hessian function.